src/index.js

```diff
@@ -207,10 +208,26 @@ function sanitizeHtml(html, options, _recursing) {
               }
             }
             if (name === 'iframe' && a === 'src') {
-              var whitelistedUrls = options.allowedIframeDomains.filter(function(url) {
-                return value.includes(url);
-              })
-              if (!whitelistedUrls.length) {
+              try {
+                var parsed = url.parse(value)
+                var whitelistedDomains = []
+                each(parsed, function(val, prop) {
+                  if (prop === 'hostname') {
+                    whitelistedDomains = options.allowedIframeDomains.filter(function(domain) {
```

Comment:
`find` is faster than `filter` here and less weird because you can use it directly in an `if` statement.

Reply:
Changed.